glusterd2/commands/devices/adddevice.go

```diff
@@ -0,0 +1,110 @@
+package devicecommands
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/pborman/uuid"
+	"net/http"
+
+	"github.com/coreos/etcd/clientv3"
+	device "github.com/gluster/glusterd2/glusterd2/device"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/store"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/pkg/api"
+	log "github.com/sirupsen/logrus"
+)
+
+func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URLi
+	ctx := r.Context()
```

Comment:
Derive a logger from the context here and use it consistently through-out and log errors.